c10/util/AbortHandler.h

```diff
@@ -0,0 +1,19 @@
+#include <c10/util/Backtrace.h>
+#include <cstdlib>
+#include <iostream>
+#include <exception>
+
+namespace c10 {
+namespace detail {
+void terminate_handler() {
+  std::cout << "Unhandled exception caught in c10/util/AbortHandler.h" << std::endl;
+  auto backtrace = get_backtrace();
+  std::cout << backtrace << std::endl;
+  std::abort();
+}
+}
+
+void set_terminate_handler() {
+  std::set_terminate(detail::terminate_handler);
```

Comment:
Can we check if terminate handler is already set and print working if that is the case (to avoid overriding terminate handler that might be set by other users of the library)

Reply:
Sure. Will do.